Changes in module 'nf-core/bedtools/coverage'
--- modules/nf-core/bedtools/coverage/main.nf
+++ modules/nf-core/bedtools/coverage/main.nf
@@ -1,5 +1,4 @@
 process BEDTOOLS_COVERAGE {
-    tag "$meta.id"
     label 'process_medium'
 
     conda "bioconda::bedtools=2.30.0"
@@ -12,14 +11,14 @@
     path genome_file
 
     output:
-    tuple val(meta), path("*.bed"), emit: bed
+    tuple val(meta), path("*.cov"), emit: bed
     path "versions.yml"           , emit: versions
 
     when:
     task.ext.when == null || task.ext.when
 
     script:
-    def args = task.ext.args ?: ''
+    def args = task.ext.args ? task.ext.args + ' -counts' : '-counts'
     def prefix = task.ext.prefix ?: "${meta.id}"
     def reference = genome_file ? "-g ${genome_file} -sorted" : ""
     """
@@ -29,7 +28,7 @@
         $reference \\
         -a $input_A \\
         -b $input_B \\
-        > ${prefix}.bed
+        > ${prefix}.cov
 
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":

************************************************************
