Changes in module 'nf-core/freebayes'
--- modules/nf-core/freebayes/main.nf
+++ modules/nf-core/freebayes/main.nf
@@ -1,14 +1,15 @@
 process FREEBAYES {
     tag "$meta.id"
-    label 'process_single'
+    label 'process_medium'
 
-    conda "bioconda::freebayes=1.3.6"
+    conda (params.enable_conda ? "bioconda::freebayes=1.3.5" : null)
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/freebayes:1.3.6--hbfe0e7f_2' :
-        'quay.io/biocontainers/freebayes:1.3.6--hbfe0e7f_2' }"
+        'https://depot.galaxyproject.org/singularity/freebayes:1.3.5--py38ha193a2f_3' :
+        'quay.io/biocontainers/freebayes:1.3.5--py38ha193a2f_3' }"
 
     input:
-    tuple val(meta), path(input_1), path(input_1_index), path(input_2), path(input_2_index), path(target_bed)
+    tuple val(meta), path(bam), path(bai), path(target_bed)
+    val targets_file
     path fasta
     path fasta_fai
     path samples
@@ -25,12 +26,11 @@
     script:
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
-    def input            = input_2        ? "${input_1} ${input_2}"        : "${input_1}"
     def targets_file     = target_bed     ? "--target ${target_bed}"       : ""
     def samples_file     = samples        ? "--samples ${samples}"         : ""
     def populations_file = populations    ? "--populations ${populations}" : ""
     def cnv_file         = cnv            ? "--cnv-map ${cnv}"             : ""
-
+    def interval         = meta.interval  ? '_' + meta.interval            : ''
     """
     freebayes \\
         -f $fasta \\
@@ -39,13 +39,11 @@
         $populations_file \\
         $cnv_file \\
         $args \\
-        $input > ${prefix}.vcf
-
-    bgzip ${prefix}.vcf
-
+        $bam | \\
+    bgzip -c > ${prefix}${interval}.vcf.gz
     cat <<-END_VERSIONS > versions.yml
     "${task.process}":
         freebayes: \$(echo \$(freebayes --version 2>&1) | sed 's/version:\s*v//g' )
     END_VERSIONS
     """
-}
+}
************************************************************
