Changes in module 'nf-core/bcftools/sort'
--- modules/nf-core/bcftools/sort/main.nf
+++ modules/nf-core/bcftools/sort/main.nf
@@ -25,11 +25,11 @@
                     args.contains("--output-type z") || args.contains("-Oz") ? "vcf.gz" :
                     args.contains("--output-type v") || args.contains("-Ov") ? "vcf" :
                     "vcf"
-
+    def interval = meta.interval ?: ''
     """
     bcftools \\
         sort \\
-        --output ${prefix}.${extension} \\
+        --output ${prefix}_${interval}_sort.${extension} \\
         $args \\
         $vcf
 

************************************************************
